drivers/adc/adc_npcx.c

```diff
@@ -291,6 +383,277 @@ static int adc_npcx_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
+static void adc_npcx_set_repetitive(const struct device *dev, int chnsel,
+				    uint8_t enable)
+{
+	struct adc_reg *const inst = HAL_INSTANCE(dev);
+	struct adc_npcx_data *const data = dev->data;
+	struct adc_npcx_threshold_data *const t_data = data->threshold_data;
+
+	/* Stop ADC conversion */
+	inst->ADCCNF |= BIT(NPCX_ADCCNF_STOP);
+
+	if (enable) {
+		/* Turn on ADC */
+		inst->ADCCNF |= BIT(NPCX_ADCCNF_ADCEN);
+		/* Set ADC conversion code to SW conversion mode */
+		SET_FIELD(inst->ADCCNF, NPCX_ADCCNF_ADCMD_FIELD,
+			  NPCX_ADC_SCAN_CONVERSION_MODE);
+		/* Update number of channel to be converted */
+		inst->ADCCS |= BIT(chnsel);
+		/* Set conversion type to repetitive (runs continuously) */
+		inst->ADCCNF |= BIT(NPCX_ADCCNF_ADCRPTC);
+
+		t_data->repetitive_channels |= BIT(chnsel);
+		/* Start conversion */
+		inst->ADCCNF |= BIT(NPCX_ADCCNF_START);
+	} else {
+		inst->ADCCS &= ~BIT(chnsel);
+
+		t_data->repetitive_channels &= ~BIT(chnsel);
+		if (!t_data->repetitive_channels) {
+			/* No thesholdd active left, disable repetitive mode */
+			inst->ADCCNF &= ~BIT(NPCX_ADCCNF_ADCRPTC);
+			/* Turn off ADC */
+			inst->ADCCNF &= ~BIT(NPCX_ADCCNF_ADCEN);
+		} else {
+			/* Start conversion again */
+			inst->ADCCNF |= BIT(NPCX_ADCCNF_START);
+		}
+	}
+}
+
+int adc_npcx_threshold_ctrl_set_param(const struct device *dev,
+				      const uint8_t th_sel,
+				      const struct adc_npcx_threshold_control_t
+				      *control)
+{
+	struct adc_npcx_data *const data = dev->data;
+	struct adc_npcx_threshold_data *const t_data = data->threshold_data;
+	struct adc_npcx_threshold_control *const t_ctrl =
+					&t_data->control[th_sel];
```

Comment:
The naming used here is a bit confusing. Maybe the `adc_npcx_threshold_control_t` structure could be renamed to `adc_npcx_threshold_param` (and consist of `enum adc_npcx_threshold_param_type type` and `uint32_t val` fields)?

Reply:
Sounds good, it would make more sense given function name `adc_npcx_threshold_ctrl_set_param`

Reply:
Done